conjurapi/authn.go

```diff
@@ -116,6 +138,43 @@ func (c *Client) authenticate(loginPair authn.LoginPair) (*http.Response, error)
 	return c.httpClient.Do(req)
 }
 
+func (c *Client) OidcAuthenticate(code, nonce, code_verifier string) ([]byte, error) {
+	req, err := c.OidcAuthenticateRequest(code, nonce, code_verifier)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := response.DataResponse(res)
+
+	if err == nil {
+		c.cacheOidcToken(resp)
+	}
```

Comment:
I might be missing the forest for the trees, but this doesn't seem specific to the API project, and instead could be the responsibility of the consuming client.

I can imagine a Go client that chooses to store the access token elsewhere (like Conjur UI, which uses an encrypted session cookie) and doesn't want/need the token saved to file elsewhere.

Reply:
So this is complicated. @doodlesbykumbi and I have been discussing whether the netrc storage should happen in the CLI or API. See this comment: https://github.com/cyberark/conjur-api-go/blob/52563e2e91c597c1984d133d629703184166e3e6/conjurapi/authn.go#L244-L245.

In the meantime, the practical reason I took this approach is that the CLI doesn't have access to the access token - it stays in the API library, and I didn't think it was such a great idea to change that. I'm open to more discussion though.

Reply:
Yup, I'm definitely remembering this topic now.

I'm concerned that if a user is only dealing with the API's public interface, they'll notice the `OidcAuthenticate` function returns an access token, and the token being written to a file could be either unintended or invisible.

Or maybe I'm thinking about this the wrong way, and the API is using the cached token to authenticate HTTP calls, so what is really "unintended" is returning the access token to a user at all!

Maybe this means that returning an access token and caching one should be two distinct function calls.

Reply:
That's an interesting point. In truth, we could make the `Authenticate()` and `OidcAuthenticate()` methods private, as far as the CLI is concerned. I suppose the issue then would be for clients that want to use the API in a different way.
Maybe the best solution would be to make a separate function for caching the token, but I would argue that by calling `OidcAuthenticate()` instead of `Authenticate()` it's implicit that this flow (including caching) should be used. OIDC generally can't be used for automation as the whole advantage it has over regular authn is that it requires a second factor, ie user interaction.